Std/Data/BitVec/Lemmas.lean

```diff
@@ -155,6 +157,41 @@ theorem truncate_succ (x : BitVec w) :
     have j_lt : j.val < i := Nat.lt_of_le_of_ne (Nat.le_of_succ_le_succ j.isLt) j_eq
     simp [j_eq, j_lt]
 
+/-! ### cast -/
+
+@[simp]
+theorem add_cast {x y : BitVec w} :  (x + y).toNat = (x.toNat + y.toNat) % 2^w := rfl
+@[simp]
+theorem sub_cast {x y : BitVec w} : (x - y).toNat = (x.toNat + (2^w - y.toNat)) % 2^w := rfl
+
+@[simp]
+theorem neg_cast {x : BitVec w} : (-x).toNat = (2^w- x.toNat) % 2^w := by
+  rw [← Nat.zero_add (2^w - x.toNat)]; rfl
+
+@[simp]
+theorem xor_cast {x y: BitVec w} : (x ^^^ y).toNat = x.toNat ^^^ y.toNat:= rfl
+
+/-! ### ring properties -/
```

Comment:
Let's try to be a bit less mathy in Std. Besides, since there are no mul lemmas (yet), "ring" is a bit too ambitious for now. Split these lemmas in sections for `add`, `sub`, etc.